### PR TITLE
Feature/forward ports #724 #739 & #740 [6460]

### PR DESCRIPTION
**Pull request by LuisGP**
This PR brings to 1.9.x branch the PRs #724 #739 and #740.

### CODE REVIEW

File: include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h [hunk 1]
```diff
@@ -137,30 +155,36 @@ class DiscoverySettings
...
      * as well as to all Multicast ports.
      */
-    Duration_t leaseDuration_announcementperiod;
+    Duration_t leaseDuration_announcementperiod = { 40, 0 };
+
+    //!Initial announcements configuration
+    InitialAnnouncementConfig initial_announcements;
+
+    //!Set to true to avoid multicast traffic on builtin endpoints
+    bool avoid_builtin_multicast = false;
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h]: I also miss the change that discards multicast locators when matching builtin endpoints

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/508/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/454/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/504/)

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 2]
```diff
@@ -177,6 +181,7 @@ TEST_F(XMLProfileParserTests, XMLParserDefaultParcipantProfile)
     EXPECT_EQ(builtin.discovery_config.leaseDuration, c_TimeInfinite);
     EXPECT_EQ(builtin.discovery_config.leaseDuration_announcementperiod.seconds, 10);
     EXPECT_EQ(builtin.discovery_config.leaseDuration_announcementperiod.nanosec, 333u);
+    EXPECT_FALSE(builtin.avoid_builtin_multicast);
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp]: Should we add initial_announcements checks here?

File: test/unittest/xmlparser/test_xml_profiles_rooted.xml [hunk 3]
```diff
@@ -51,6 +51,7 @@
                             <sec>10</sec>
                             <nanosec>333</nanosec>
                         </leaseAnnouncement>
+                        <avoid_builtin_multicast>false</avoid_builtin_multicast>
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/test_xml_profiles_rooted.xml]: Should we add initial_announcements configuration here?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/518/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/463/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/514/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/545/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/480/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/541/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/559/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/487/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/555/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/567/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/494/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/563/)

**Comment by MiguelBarro** [comment]
I've just run discovery server against this fastrtps branch and all works fine. 👍

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/579/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/505/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/575/)

**Comment by raquelalvarezbanos** [comment]
Mac failure seems unrelated, so merging this in.

**Pull request merged**